packages/server-wallet/src/protocols/__test__/ledger-manager.test.ts

```diff
@@ -62,660 +32,735 @@ beforeEach(async () => {
-      };
-      expect(payload).toMatchObject({
-        requests: [serializeRequest(expectedProposeLedgerUpdate)],
-      });
-    });
+describe('as follower', () => {
+  describe('in the agreement state', () => {
+    it(
+      `marks included requests as succeeded`,
+      testLedgerCrank({
+        as: 'follower',
+        before: {
+          agreed: {turn: 5, bals: {a: 10, b: 10, c: 2}},
+          requests: [
+            ['defund', 'd', 1, 1, 'pending'], // d is not there, so this succeeded
+            ['fund', 'c', 2, 0, 'pending'], // this also succeeded
+          ],
+        },
+        after: {
+          agreed: {turn: 5, bals: {a: 10, b: 10, c: 2}},
+          requests: [
+            ['defund', 'd', 1, 1, 'succeeded', {missedOps: 0, lastSeen: 5}],
+            ['fund', 'c', 2, 0, 'succeeded', {missedOps: 0, lastSeen: 5}],
+          ],
+        },
+      })
+    );
+    it(
+      `marks inconsistent states`,
+      testLedgerCrank({
+        as: 'follower',
+        before: {
+          agreed: {turn: 5, bals: {a: 10, b: 10, c: 2, d: 1}},
+          requests: [
+            ['defund', 'd', 1, 1, 'queued'], // amount doesn't match with d's total
+            ['fund', 'c', 2, 1, 'queued'], // c is funded, but with a different amt
+          ],
+        },
+        after: {
+          agreed: {turn: 5, bals: {a: 10, b: 10, c: 2, d: 1}},
+          requests: [
+            ['defund', 'd', 1, 1, 'inconsistent', {missedOps: 0, lastSeen: 5}],
+            ['fund', 'c', 2, 1, 'inconsistent', {missedOps: 0, lastSeen: 5}],
+          ],
+        },
+      })
+    );
+    it(
+      `will cancel requests that haven't been included`,
+      testLedgerCrank({
+        as: 'follower',
+        before: {
+          agreed: {turn: 5, bals: {a: 10, b: 10}},
+          requests: [
+            ['fund', 'c', 2, 1, 'queued'],
+            ['defund', 'c', 2, 1, 'queued'],
+          ],
+        },
+        after: {
+          agreed: {turn: 5, bals: {a: 10, b: 10}},
+          requests: [
+            ['fund', 'c', 2, 1, 'cancelled', {missedOps: 0, lastSeen: 5}],
+            ['defund', 'c', 2, 1, 'cancelled', {missedOps: 0, lastSeen: 5}],
+          ],
+        },
+      })
+    );
+    it(
+      `won't cancel requests that have been included`,
+      testLedgerCrank({
+        as: 'follower',
+        before: {
+          agreed: {turn: 5, bals: {a: 10, b: 10, c: 3}},
+          requests: [
+            ['fund', 'c', 2, 1, 'pending'], // has been included
+            ['defund', 'c', 2, 1, 'queued'],
+          ],
+        },
+        after: {
+          agreed: {turn: 5, bals: {a: 10, b: 10, c: 3}},
+          requests: [
+            ['fund', 'c', 2, 1, 'succeeded', {missedOps: 0, lastSeen: 5}],
+            ['defund', 'c', 2, 1, 'queued', {missedOps: 0, lastSeen: 5}],
+          ],
+        },
+      })
+    );
+    // TODO: it will mark a mismatched cancellation as inconsistent
```

Comment:
Does this still apply?

Reply:
I think we probably don't need this behaviour at this point.